packages/lib/src/vis/hooks.ts

```diff
@@ -101,3 +102,28 @@ export function useCustomColors(
 
   return [colors, refCallback];
 }
+
+export function useWorldToHtml(): (point: Vector2 | Vector3) => Vector2 {
+  const camera = useThree((state) => state.camera);
+  const size = useThree((state) => state.size);
+
+  const cameraToHtmlMatrix = useMemo(() => {
+    const { width, height } = size;
+
+    const matrix = new Matrix4().makeScale(width / 2, -height / 2, 1);
+    // Account for shift of (0,0) position (center for camera, top-left for HTML)
+    matrix.setPosition(width / 2, height / 2);
+    return matrix;
+  }, [size]);
+
+  useFrameRendering();
+
+  return useCallback(
+    (point: Vector2 | Vector3) => {
+      const cameraPoint = new Vector3(point.x, point.y, 0).project(camera);
+      const htmlPoint = cameraPoint.clone().applyMatrix4(cameraToHtmlMatrix);
+      return new Vector2(htmlPoint.x, htmlPoint.y);
+    },
+    [camera, cameraToHtmlMatrix]
+  );
```

Comment:
I don't think there's a need for all this memoisation, since we use `useFrameRendering` anyway -- memoisation of functions like this is more to prevent triggering extra renders -- but here we already re-render on every frame.